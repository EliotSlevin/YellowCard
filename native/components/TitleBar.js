import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import styles from '../styles'

const titleBarStyle = StyleSheet.create({
  container: { padding: 7.5, height: '22%', backgroundColor: '#fffd86' },
  title: { marginTop: 7.5, fontSize: 40, fontWeight: '400' },
  subTitle: {marginTop: 7.5}
})

export default class TitleBar extends React.Component {
  render() {
    const title = this.props.title
    const subTitle = this.props.subTitle

    return (
      <View style={[titleBarStyle.container]}>
        <Text style={[titleBarStyle.title, styles.medium40]}>{title}</Text>
        <Text style={[titleBarStyle.subTitle, styles.regular17]}>{subTitle}</Text>
      </View>
    )
  }
}