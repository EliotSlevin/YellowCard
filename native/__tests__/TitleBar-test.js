import 'react-native'
import React from 'react'
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

import TitleBar from '../components/TitleBar'

test('renders correctly', () => {
  const tree = renderer.create(<TitleBar />).toJSON();
  expect(tree).toMatchSnapshot()
});
