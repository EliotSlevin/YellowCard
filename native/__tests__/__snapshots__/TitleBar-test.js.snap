// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fffd86",
        "height": "22%",
        "padding": 7.5,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 40,
          "fontWeight": "400",
          "marginTop": 7.5,
        },
        Object {
          "fontFamily": "SFUIText-Medium",
          "fontSize": 40,
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "marginTop": 7.5,
        },
        Object {
          "fontFamily": "SFUIText-Regular",
          "fontSize": 17,
        },
      ]
    }
  />
</View>
`;
